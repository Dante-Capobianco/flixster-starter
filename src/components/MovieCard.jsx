// Take in movie details as props then format it in return statement here
// Use movie-card class