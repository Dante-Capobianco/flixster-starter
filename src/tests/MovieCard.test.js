import renderer from "react-test-renderer";
import MovieCard from "../components/MovieCard";

it("should format movie tile correctly", () => {
  const tree = renderer
    .create(
      <MovieCard
        key={123}
        src="https://image.tmdb.org/t/p/w500//lbimIPTVsSlnmqSW5ngEsUxtHLM.jpg"
        alt="Movie Poster Image"
        title="Movie Title"
        isMovieFavorited={false}
        isMovieWatched={false}
        vote_average={"7.95"}
        setSelectedMovieData={() => {}}
        movieData={{id: '123'}}
        favoriteMovies={[]}
        setFavoriteMovies={() => {}}
        watchedMovies={[]}
        setWatchedMovies={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
