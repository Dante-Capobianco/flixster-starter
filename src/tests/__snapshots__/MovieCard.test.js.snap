// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should format movie tile correctly 1`] = `
<article
  className="movie-card"
  onClick={[Function]}
>
  <img
    alt="Movie Poster Image"
    className="movie-card-poster"
    src="https://image.tmdb.org/t/p/w500//lbimIPTVsSlnmqSW5ngEsUxtHLM.jpg"
  />
  <h2
    className="movie-card-title"
  >
    Movie Title
  </h2>
  <section
    className="rating-fav-watch-container"
  >
    <h3
      className="movie-card-rating"
    >
      Rating: 
      7.95
    </h3>
    <span
      class=" heart material-symbols-outlined"
    >
      favorite
    </span>
    <span
      class=" eye material-symbols-outlined"
    >
      visibility
    </span>
  </section>
</article>
`;
